### PR TITLE
Fly proxy ctrl

### PR DESCRIPTION
**Pull request by senyosimpson**
Adds the `fly_proxy` machines option which is used to configure behaviour of the proxy specifically for machines

### CODE REVIEW

**Comment by dangra** [comment]
@senyosimpson does it supersede #1871  ?

**Comment by senyosimpson** [comment]
> @senyosimpson does it supersede #1871 ?

yeah! i'll need a deprecation plan for that

**Comment by dangra** [comment]
> > @senyosimpson does it supersede #1871 ?
> 
> yeah! i'll need a deprecation plan for that

it's not merged, we can just close it. I guess the deprecation plan is more for flaps/flyd side.

**Comment by senyosimpson** [comment]
> > > @senyosimpson does it supersede #1871 ?
> > 
> > 
> > yeah! i'll need a deprecation plan for that
> 
> it's not merged, we can just close it. I guess the deprecation plan is more for flaps/flyd side.

ah yep correct

File: api/machine_types.go [hunk 1]
```diff
@@ -379,6 +379,11 @@ type MachineServiceConcurrency struct {
 	SoftLimit int    `json:"soft_limit,omitempty" toml:"soft_limit,omitempty"`
 }
 
+type MachineFlyProxy struct {
+	AutostartMachine bool `json:"autostart_machine"`
+	AutostopMachine  bool `json:"autostop_machine"`
```
Comment (dangra) [inline api/machine_types.go:384]: same thing here, `*bool` is the only way we can know when a flag is not set and left to server defaults.

**Review by dangra — APPROVED** [review]
pre-approving in case I am not around

**Pull request merged**